src/main/java/org/folio/service/inventory/InventoryItemManager.java

```diff
@@ -370,6 +392,29 @@ public Future<List<String>> createMissingPhysicalItems(CompositePoLine compPOL,
       });
   }
 
+  /**
+   * Recreates Items in the inventory based on the PO line data with the same Item id.
+   *
+   * @param compPOL  PO line to create Instance Record for
+   * @param piece    base piece to build item
+   * @param quantity expected number of items to create
+   * @return id of newly created Instance Record
+   */
+  public Future<List<String>> recreateMissingPhysicalItems(CompositePoLine compPOL, Piece piece, int quantity,
+                                                           RequestContext requestContext) {
+    if (quantity <= 0) {
+      return Future.succeededFuture(List.of());
+    }
+    String holdingId = piece.getHoldingId();
+    return buildPhysicalItemRecordJsonObject(compPOL, holdingId, requestContext)
+      .compose(item -> {
+        InventoryUtils.updateItemWithPieceFields(item, piece);
+        item.put(ID, piece.getItemId());
+        logger.debug("Posting {} physical item(s) for PO Line with '{}' id and '{}' item id", quantity, compPOL.getId(), piece.getItemId());
```

Comment:
let's change log level to info and for similar new places. This is useful information to see in logs in any case.

Reply:
by the way, we already have methods: createMissingPhysicalItems, createMissingElectronicItems.
So they differ only by this line: item.put(ID, piece.getItemId());
As one of potential solution you can add new parameter itemId to them and previous callers will send null or random uuids.

Reply:
> by the way, we already have methods: createMissingPhysicalItems, createMissingElectronicItems. So they differ only by this line: item.put(ID, piece.getItemId()); As one of potential solution you can add new parameter itemId to them and previous callers will send null or random uuids.

Yeah, I originally thought doing so, but that would be bit disruptive, I will make some changes now to reusable the existing 2 create methods